zebra-state/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+#![doc(html_logo_url = "https://www.zfnd.org/images/zebra-icon.png")]
+#![doc(html_root_url = "https://doc.zebra.zfnd.org/zebra_storage")]
+use futures::prelude::*;
+use std::{
+    collections::HashMap,
+    error::Error,
+    future::Future,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+use tower::{buffer::Buffer, Service};
+use zebra_chain::block::{Block, BlockHeaderHash};
+
+pub enum Request {
+    // TODO(jlusby): deprecate in the future based on our validation story
+    AddBlock { block: Arc<Block> },
+    GetBlock { hash: BlockHeaderHash },
+}
+
+pub enum Response {
+    Ok,
+    Block { block: Arc<Block> },
```

Comment:
following on to this, there should be an issue to replace `Box<Block>` with `Arc<Block>`